app/experimenter/experiments/serializers.py

```diff
@@ -616,36 +714,65 @@ def update(self, instance, validated_data):
         return instance
 
 
-class PrefValidationMixin(object):
+class ExperimentDesignMultiPrefSerializer(ExperimentDesignBaseSerializer):
+    type = serializers.CharField()
+    is_multi_pref = serializers.BooleanField()
+    variants = ExperimentDesignBranchMultiPrefSerializer(many=True)
 
-    def validate_pref(self, pref_type, pref_value, value):
-        if pref_type == "integer":
-            try:
-                int(pref_value)
-            except ValueError:
-                return {value: "The pref value must be an integer."}
+    class Meta:
+        model = Experiment
+        fields = ("type", "is_multi_pref", "variants")
 
-        if pref_type == "boolean":
-            if pref_value not in ["true", "false"]:
-                return {value: "The pref value must be a boolean."}
+    def update(self, instance, validated_data):
+        variant_preferences = [
+            (v_d, v_d.pop("preferences")) for v_d in validated_data["variants"]
+        ]
 
-        if pref_type == "json string":
-            try:
-                json.loads(pref_value)
-            except ValueError:
-                return {value: "The pref value must be valid JSON."}
-        return {}
+        instance = super().update(instance, validated_data)
+        existing_pref_ids = self.get_existing_preference_ids(instance)
+        submitted_pref_ids = []
+        for variant_data, pref in variant_preferences:
+
+            variant = ExperimentVariant.objects.get(**variant_data)
+            for preference in pref:
+                preference["variant_id"] = variant.id
+                VariantPreferences(**preference).save()
+
+                if preference.get("id"):
```

Comment:
Omg it's time for the walrus operator:

    if preference_id := preference.get("id"):
        submitted_pref_ids.append(preference_id)

:O :O :O

Reply:
walrus notation isn't support by black yet, so it fails linting, or at least it was failing locally?